lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1238 @@
+    # REDCap checkbox fields have format of {question}___{answer}
+    answers = list(map(lambda k: k.replace(f"{coded_question}___", ""), answered_checkboxes))
+
+    if coded_question == 'race':
+        return race(answers)
+
+    if re.match(r'fu_[1-4]_symptoms$', coded_question):
+        return list(map(lambda a: map_symptom(a), answers))
+
+    return answers
+
+def map_vaccine(vaccine_response: str) -> Optional[bool]:
+    """
+    Maps a vaccine response to FHIR immunization status codes
+    (https://www.hl7.org/fhir/valueset-immunization-status.html)
+    """
+    vaccine_map = {
+        'yes': True,
+        'no': False,
+        'dont_know': None,
+        '': None
+    }
+
+    if vaccine_response not in vaccine_map:
+        raise UnknownVaccineResponseError(f"Unknown vaccine response «{vaccine_response}»")
+
+    return vaccine_map[vaccine_response]
+
+def vaccine(record: Any) -> Optional[dict]:
+    """
+    For a given *record*, return a questionnaire response item with the vaccine
+    response(s) encoded.
+    """
+    vaccine_status = map_vaccine(record["vaccine"])
+    if vaccine_status is None:
+        return None
+
+    answers: List[Dict[str, Any]] = [{ 'valueBoolean': vaccine_status }]
+
+    date = vaccine_date(record)
+    if vaccine_status and date:
+        answers.append({ 'valueDate': date })
+
+    return create_questionnaire_response_item('vaccine', answers)
+
+
+def vaccine_date(record: dict) -> Optional[str]:
+    """ Converts a vaccination date to 'YYYY' or 'YYYY-MM' format. """
+    year = record['vaccine_year']
+    month = record['vaccine_month']
+
+    if year == '' or year == 'dont_know':
+        return None
+
+    if month == 'dont_know':
+        return datetime.strptime(year, '%Y').strftime('%Y')
+
+    return datetime.strptime(f'{month} {year}', '%B %Y').strftime('%Y-%m')
+
+
+def questionnaire_response(record: dict,
+                           question_categories: Dict[str, list],
+                           patient_reference: dict,
+                           encounter_reference: dict,
+                           retain_empty_string: bool = False) -> Optional[dict]:
+    """
+    Provided a dictionary of *question_categories* with the key being the value
+    type and the value being a list of field names, return a FHIR
+    Questionnaire Response resource entry.
+    Set retain_empty_string to True to include empty string values for items.
+    """
+    def build_questionnaire_items(question: str) -> Optional[dict]:
+        return questionnaire_item(record, question, category, retain_empty_string)
+
+    items: List[dict] = []
+    for category in question_categories:
+        category_items = list(map(build_questionnaire_items, question_categories[category]))
+        for item in category_items:
+            if item:
+                items.append(item)
+
+    vaccine_item = vaccine(record)
+    if vaccine_item:
+        items.append(vaccine_item)
```

Comment:
This is adding the vaccine question item to both the computed questionnaire and the enrollment questionnaire. I think we just want it the enrollment one?

Reply:
That was a really great embedded catch. We do only want the vaccine item for the enrollment question. I've added a parameter and set it accordingly in the latest commit. I tested this locally and confirmed that I get vaccine only for the enrollment questionnaire. The `questionnaire_response` function's parameters are not very elegant, but they work.